src/grpcbox_client_stream.erl

```diff
@@ -13,15 +13,16 @@
 
 -include("grpcbox.hrl").
 
--define(headers(Scheme, Host, Path, Encoding, MessageType, MD), [{<<":method">>, <<"POST">>},
-                                                                 {<<":path">>, Path},
-                                                                 {<<":scheme">>, Scheme},
-                                                                 {<<":authority">>, Host},
-                                                                 {<<"grpc-encoding">>, Encoding},
-                                                                 {<<"grpc-message-type">>, MessageType},
-                                                                 {<<"content-type">>, <<"application/grpc+proto">>},
-                                                                 {<<"user-agent">>, <<"grpc-erlang/0.9.2">>},
-                                                                 {<<"te">>, <<"trailers">>} | MD]).
+-define(protected_headers, [<<"content-type">>, <<"te">>]).
+-define(pseudoheaders(Path, Scheme, Authority), [{<<":method">>, <<"POST">>},
+                                                 {<<":path">>, Path},
+                                                 {<<":scheme">>, Scheme},
+                                                 {<<":authority">>, Authority}]).
+-define(headers(Encoding, MessageType, MD), MD ++ [{<<"grpc-encoding">>, Encoding},
+                                                   {<<"grpc-message-type">>, MessageType},
+                                                   {<<"content-type">>, <<"application/grpc+proto">>},
+                                                   {<<"user-agent">>, <<"grpc-erlang/0.9.2">>},
+                                                   {<<"te">>, <<"trailers">>}]).
```

Comment:
you may want to wrap the expression in parentheses because the `++` precedence rules could be funny in the long run compared to `(MD ++ List)` which will always be a unit.